Harmony/Tools/AccessTools.cs

```diff
@@ -836,14 +834,23 @@ public static object[] ActualParameters(MethodBase method, object[] inputs)
 		/// </typeparam>
 		/// <param name="fieldName">The name of the field</param>
 		/// <returns>A readable/assignable <see cref="FieldRef{T,F}"/> delegate</returns>
+		/// <remarks>
+		/// <para>
+		/// For backwards compatibility, there is no class constraint on <typeparamref name="T"/>.
+		/// Instead, the non-value-type check is done at runtime within the method.
+		/// </para>
+		/// </remarks>
 		///
-		public static FieldRef<T, F> FieldRefAccess<T, F>(string fieldName) where T : class
+		public static FieldRef<T, F> FieldRefAccess<T, F>(string fieldName)
 		{
 			if (fieldName is null)
 				throw new ArgumentNullException(nameof(fieldName));
 			try
 			{
-				return FieldRefAccessInternal<T, F>(GetInstanceField(typeof(T), fieldName), needCastclass: false);
+				var delegateInstanceType = typeof(T);
+				if (delegateInstanceType.IsValueType)
+					throw new ArgumentException("T (FieldRefAccess instance type) must not be a value type");
```

Comment:
Will this create an argument exception inside an argument exception?

Reply:
Yes, for the same reason that other exceptions are thrown in these try blocks - the outer exception shows all the argument values, including generic type args.